tasks/kmt.py

```diff
@@ -1072,12 +1105,14 @@ def build_layout(ctx, domains, layout: str, verbose: bool):
         todo: DependenciesLayout = cast('DependenciesLayout', json.load(lf))
 
     for d in domains:
+        info(f"[+] apply layout to vm {d.name}")
         mkdir = []
         for dirs in todo["layout"]:
             mkdir.append(f"mkdir -p {dirs} &&")
 
-        cmd = ' '.join(mkdir)
-        d.run_cmd(ctx, cmd.rstrip('&'), verbose)
+        if len(mkdir) > 0:
+            cmd = ' '.join(mkdir)
+            d.run_cmd(ctx, cmd.rstrip('&'), verbose)
```

Comment:
Suggestion `cmd = ' && '.join(f'mkdir -p {dir}' for dir in todo["layout"])`, then you can check for `len(cmd) > 0`